bson/marshal_test.go

```diff
@@ -267,3 +268,35 @@ func TestCachingEncodersNotSharedAcrossRegistries(t *testing.T) {
 		})
 	})
 }
+
+func TestNullBytes(t *testing.T) {
+	t.Run("element keys", func(t *testing.T) {
+		doc := D{{"a\x00", "foobar"}}
```

Comment:
I'm pretty sure it's safe, but to double check, is there a risk that marshalling other types (aside from `bson.D`) goes through another code path? The original report marshals a struct with a `map[string][]string`.

Reply:
All types will go through the same `ValueWriter` functionality, so it shouldn't matter.